src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -528,5 +528,53 @@ private static Vector<byte> GetVector(byte vectorByte)
                                                        0x01ul << 48) + 1;
         private const ulong byteBroadcastToUlong = ~0UL / Byte.MaxValue;
         private const ulong filterByteHighBitsInUlong = (byteBroadcastToUlong >> 1) | (byteBroadcastToUlong << (sizeof(ulong) * 8 - 1));
+
+
+        /// <summary>
+        /// Determines whether the current span is a slice of the supplied span
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsSliceOf<T>(this Span<T> childSpan, Span<T> parentSpan) where T : struct
+        {
+            int ignored;
+            return childSpan.Length <= parentSpan.Length // avoid pointer work if trivially false
+                && IsSliceOf(ref childSpan.DangerousGetPinnableReference(), childSpan.Length,
+                ref parentSpan.DangerousGetPinnableReference(), parentSpan.Length, out ignored);
+        }
+
+        /// <summary>
+        /// Determines whether the current span is a slice of the supplied span
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsSliceOf<T>(this Span<T> childSpan, Span<T> parentSpan, out int start) where T : struct
+        {
+            start = -1;
+            return childSpan.Length <= parentSpan.Length // avoid pointer work if trivially false
+                && IsSliceOf(ref childSpan.DangerousGetPinnableReference(), childSpan.Length,
+                ref parentSpan.DangerousGetPinnableReference(), parentSpan.Length, out start);
+        }
+
+        private static unsafe bool IsSliceOf<T>(ref T child, int childCount, ref T parent, int parentCount, out int start) where T : struct
+        {
+            // compare the byte addresses for validity
+            var childStart = Unsafe.AsPointer(ref child);
+            var childEnd = Unsafe.AsPointer(ref Unsafe.Add(ref child, childCount));
+            var parentStart = Unsafe.AsPointer(ref parent);
```

Comment:
No problem, you should be able to do this with something like (just to give an overview example of what I thought):
```csharp
IntPtr byteOffset = Unsafe.ByteOffset(parent, child);
// If child before parent, then not slice of
if (byteOffset < 0)
{ return false; }
var offset = byteOffset / Unsafe.SizeOf<T>();
// If offset in items is longer than length, then child is after parent
if (offset >= parent.Length)
{ return false; }

// And so forth...

```
For the arithmetic and comparisons you need to do custom 32/64-bit handling or use a proper `nint` type from e.g. https://github.com/DotNetCross/NativeInts (shameless plug for my own code ;))